third_party/tektoncd-cli/pkg/cmd/taskrun/log_reader.go

```diff
@@ -0,0 +1,268 @@
+	Follow   bool
+	AllSteps bool
+
+	// BlacklistSteps prvents steps from being displated when AllSteps is
+	// enabled.
+	//
+	// NOTE: BlacklistSteps doesn't exist in the normal tektoncd code.
+	// However, istio is SOO chatty that it is distracting. So I want to be
+	// able to mute it.
+	BlacklistSteps []string
+}
+
+const (
+	msgTRNotFoundErr = "Unable to get Taskrun"
+)
+
+func (lr *LogReader) Read() (<-chan Log, <-chan error, error) {
+	tkn := lr.Clients.Tekton
+	tr, err := tkn.TektonV1alpha1().TaskRuns(lr.Ns).Get(lr.Run, metav1.GetOptions{})
+	if err != nil {
+		return nil, nil, fmt.Errorf("%s : %s", msgTRNotFoundErr, err)
+	}
+
+	lr.formTaskName(tr)
+
+	return lr.readLogs(tr)
+}
+
+func (lr *LogReader) readLogs(tr *v1alpha1.TaskRun) (<-chan Log, <-chan error, error) {
+	if lr.Follow {
+		return lr.readLiveLogs(tr)
+	}
+	return lr.readAvailableLogs(tr)
+}
+
+func (lr *LogReader) formTaskName(tr *v1alpha1.TaskRun) {
+	if lr.Task != "" {
+		return
+	}
+
+	if name, ok := tr.Labels["tekton.dev/pipelineTask"]; ok {
+		lr.Task = name
+		return
+	}
+
+	if tr.Spec.TaskRef != nil {
+		lr.Task = tr.Spec.TaskRef.Name
+		return
+	}
+
+	lr.Task = fmt.Sprintf("Task %d", lr.Number)
+}
+
+func (lr *LogReader) readLiveLogs(tr *v1alpha1.TaskRun) (<-chan Log, <-chan error, error) {
+	var (
+		podName = tr.Status.PodName
+		kube    = lr.Clients.Kube
+	)
+
+	p := pods.New(podName, lr.Ns, kube, lr.Streamer)
+	pod, err := p.Wait()
+	if err != nil {
+		return nil, nil, errors.Wrap(err, fmt.Sprintf("task %s failed", lr.Task))
+	}
+
+	steps := filterSteps(pod, lr.AllSteps, lr.BlacklistSteps)
+	logC, errC := lr.readStepsLogs(steps, p, lr.Follow)
+	return logC, errC, err
+}
+
+func (lr *LogReader) readAvailableLogs(tr *v1alpha1.TaskRun) (<-chan Log, <-chan error, error) {
+	var (
+		kube    = lr.Clients.Kube
+		podName = tr.Status.PodName
+	)
+
+	if !tr.HasStarted() {
+		return nil, nil, fmt.Errorf("task %s has not hasStarted yet", lr.Task)
+	}
+
+	p := pods.New(podName, lr.Ns, kube, lr.Streamer)
+	pod, err := p.Get()
+	if err != nil {
+		return nil, nil, errors.Wrap(err, fmt.Sprintf("task %s failed", lr.Task))
+	}
+
+	steps := filterSteps(pod, lr.AllSteps, lr.BlacklistSteps)
+	logC, errC := lr.readStepsLogs(steps, p, lr.Follow)
+	return logC, errC, nil
+}
+
+func (lr *LogReader) readStepsLogs(steps []*step, pod *pods.Pod, follow bool) (<-chan Log, <-chan error) {
+	logC := make(chan Log)
+	errC := make(chan error)
+
+	go func() {
+		defer close(logC)
+		defer close(errC)
+
+		for _, step := range steps {
+			if !follow && !step.hasStarted() {
+				continue
+			}
+
+			container := pod.Container(step.container)
+			podC, perrC, err := container.LogReader(follow).Read()
+			if err != nil {
+				errC <- fmt.Errorf("error in getting logs for step %s : %s", step.name, err)
+				continue
+			}
+
+			for podC != nil || perrC != nil {
+				select {
+				case l, ok := <-podC:
+					if !ok {
+						podC = nil
+						logC <- Log{Task: lr.Task, Step: step.name, Log: "EOFLOG"}
```

Comment:
I know this is `third_party`, but it results in some ugly looking logs. can we remove this line?
```
[buildpack/detect] ======== Results ========
[buildpack/detect] fail: io.buildpacks.samples.nodejs@0.0.1
[buildpack/detect] ======== Results ========
[buildpack/detect] pass: io.buildpacks.samples.go@0.0.1
[buildpack/detect] Resolving plan... (try #1)
[buildpack/detect] Success! (1)
[buildpack/detect] EOFLOG
[buildpack/restore] Cache '/cache': metadata not found, nothing to restore
[buildpack/restore] EOFLOG
[buildpack/analyze] Image 'gcr.io/kf-source/app_sweeps2_echo2:2' not found
[buildpack/analyze] EOFLOG
[buildpack/build] Downloading go: (https://dl.google.com/go/go1.12.1.linux-amd64.tar.gz)...
[buildpack/build] Done installing go: (1.12.1).
[buildpack/build] go version go1.12.1 linux/amd64
[buildpack/build] EOFLOG
[buildpack/export] Exporting layer 'app' with SHA sha256:b6c7faa283a220674d98f4faa56235b71d986a84ac0dda4f927f84099244d764
[buildpack/export] Exporting layer 'config' with SHA sha256:074be3a09117b3ad2bbe9d6275042c3c5b7e5148db0b9685b2e9b31e564e429f
[buildpack/export] Exporting layer 'launcher' with SHA sha256:ef2cd5869b4d3995edbba8144c8c448f864f3895ea264967701a9a97547d6087
[buildpack/export] Exporting layer 'io.buildpacks.samples.go:go' with SHA sha256:f1a76d325087a29ba37fefd33ff2a521f1ab4b0b0f9f598205ecd32cfb6461d4
[buildpack/export] *** Images:
[buildpack/export]       gcr.io/kf-source/app_sweeps2_echo2:2 - succeeded
[buildpack/export] 
[buildpack/export] *** Digest: sha256:db75d19c2fd00c81125893276440b69d8b6923a8d85098ff5e25575dd7d9ede6
[buildpack/export] EOFLOG
[buildpack/cache] Caching layer 'io.buildpacks.samples.go:go' with SHA sha256:f1a76d325087a29ba37fefd33ff2a521f1ab4b0b0f9f598205ecd32cfb6461d4
[buildpack/cache] EOFLOG
```

Reply:
Yea that looks... not great. I see you have a related PR: https://github.com/tektoncd/cli/pull/442

Reply:
After talking offline, it looks like we're going to leave this alone for now.